source/simulator/solver_schemes.cc

```diff
@@ -429,9 +429,20 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::solve_newton_stokes ()
   {
+    // The values of the pressure in current_linearization_point end up being
+    // different then what they where at the end of this loop, and different
+    // from solution in a bad way for the models I tested. Therefore I first
+    // replace current_linearization_point by the solution.
```

Comment:
Before starting the nonlinear loop the current linearization point will be the extrapolation of the two available old solutions to the current time (see core.cc:1724-1738). This is part of the BDF2 time stepping scheme. I am not sure why this does not work for your test problem, but I am pretty sure that the convergence order of the scheme depends on this fact. Can you specify `a bad way`?